lib/util/parseHeaders.js

```diff
@@ -0,0 +1,29 @@
+const parseEmojis = str => {
+  const emojiData = require('markdown-it-emoji/lib/data/full.json')
+  return str.replace(/:(.+?):/g, (placeholder, key) => emojiData[key] || placeholder)
+}
+
+const unescapeHtml = html => html
+  .replace(/&quot;/g, '"')
+  .replace(/&#39;/g, '\'')
+  .replace(/&#x3A;/g, ':')
+  .replace(/&lt;/g, '<')
+  .replace(/&gt;/g, '>')
+
+const removeMarkdownToken = str => str
+  .replace(/`(.*)`/, '$1')  // ``
+  .replace(/\[(.*)\]\(.*\)/, '$1')  // []()
+  .replace(/\*\*(.*)\*\*/, '$1')  // **
+  .replace(/\*(.*)\*/, '$1')  // *
+  .replace(/_(.*)_/, '$1')  // _
+
+// put here to avoid circular references
+const compose = (...processors) => {
```

Comment:
The checking of the `processors.length` seems unnecessary. Perhaps something like this?

```js
const compose = (...processors) => {
  return processors.reduce((prev, next) => {
    return (...args) => next(prev(...args))
  }, x => x)
}
```

Or

```js
const compose = (...processors) => {
  return [
    x => x,
    ...processors
  ].reduce((prev, next) => {
    return (...args) => next(prev(...args))
  })
}
```

Reply:
Hmmm? Are you telling me how to write code? 

Code should be as easy as possible to understand, don't always think about drag racing.

If you expect me to make a change, first give redux a PR.

Reply:
@ulivz I also prefer easier to write code, but this is open source and @ycmjason is just offering what he thinks might be better code. He is not telling you how to write code and you don't have to be combative about this.

Reply:
@yyx990803 Yes, I should say convincing reasons instead of arguing for it.  

@ycmjason I'm terribly sorry about my recklessness, and I liked the style what you said above, but, after a long time, I found I always need to think back to what parameters of reduce's callback needs to accept. Until one day I read the source code of Redux, I found it's easier to write. so I'd like this style.